build/helper/metadata_add_all.py

```diff
@@ -503,6 +503,31 @@ def _add_enum_value_python_name(enum_info, config):
     return enum_info
 
 
+def fixup_enum_names(config):
+    '''Fix enum types for private enums
+
+    Now that we have all the metadata calculated, we need to fix any enum types in attributes and functions
+    where the underlying enum is private. At the time the 'python_type' was set, we hadn't yet calculated
+    whether the enum would be private or not. We couldn't because we needed to process all the functions and
+    attributes first.
+    '''
+    # Check all the functions that will be code generated
+    for f in config['functions']:
+        if config['functions'][f]['codegen_method'] != 'no':
+            for p in config['functions'][f]['parameters']:
+                if p['enum'] is not None and config['enums'][p['enum']]['codegen_method'] == 'private':
```

Comment:
Why check that it's private?
Can't we just always replace? Public enums won't change. (less logic / hidden contract)

Applies elsewhere.